lib/backend/compat/compat.go

```diff
@@ -199,6 +239,33 @@ func (c *ModelAdaptor) getProfile(k model.Key) (*model.KVPair, error) {
 	return &d, nil
 }
 
+// getBlockAffinity gets KVPair for Block. It gets the block value first,
+// then checks for `Affinity` field first, then `HostAffinity` as a backup.
+// For more details see: https://github.com/projectcalico/libcalico-go/issues/226
+func (c *ModelAdaptor) getBlockAffinity(k model.Key) (*model.KVPair, error) {
```

Comment:
I think this should just be called `getBlock`

A `blockAffinity` is a different thing :)